controller/internal/routes/identity_api_model.go

```diff
@@ -59,18 +60,27 @@ func (factory *IdentityLinkFactoryImpl) Links(entity models.Entity) rest_model.L
 	return links
 }
 
+func getTerminatorCost(v *rest_model.TerminatorCost) uint16 {
+	if v == nil {
+		return 0
+	}
+	return uint16(*v)
+}
+
 func MapCreateIdentityToModel(identity *rest_model.IdentityCreate, identityTypeId string) (*model.Identity, []*model.Enrollment) {
 	var enrollments []*model.Enrollment
 
 	ret := &model.Identity{
 		BaseEntity: models.BaseEntity{
 			Tags: identity.Tags,
 		},
-		Name:           stringz.OrEmpty(identity.Name),
-		IdentityTypeId: identityTypeId,
-		IsDefaultAdmin: false,
-		IsAdmin:        *identity.IsAdmin,
-		RoleAttributes: identity.RoleAttributes,
+		Name:                     stringz.OrEmpty(identity.Name),
+		IdentityTypeId:           identityTypeId,
+		IsDefaultAdmin:           false,
+		IsAdmin:                  *identity.IsAdmin,
+		RoleAttributes:           identity.RoleAttributes,
+		DefaultHostingPrecedence: ziti.GetPrecedenceForLabel(string(identity.DefaultHostingPrecedence)),
```

Comment:
Why? Precedence -> string -> Precedence?

Reply:
identity.DefaultHostingPrecedence is of type TerminatorPrecedence which is generated type, but is just a type alias for string.

So we're really just going string -> Precedence